### PR TITLE
python3Packages.pyiqvia: init 0.3.1

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Python3 API for IQVIA data

https://github.com/bachya/pyiqvia

This is a Home Assistant dependency.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108680` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python37Packages.pyiqvia</li>
<li>python38Packages.pyiqvia</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108680` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.pyiqvia: log <a href="https://hastebin.com/raw/ixucozigon">https://hastebin.com/raw/ixucozigon</a></li>
<li>python38Packages.pyiqvia: log <a href="https://hastebin.com/raw/himihejiza">https://hastebin.com/raw/himihejiza</a></li>
<li>python39Packages.pyiqvia: log <a href="https://hastebin.com/raw/ubegosalib">https://hastebin.com/raw/ubegosalib</a></li>
  </ul>
</details>

File: pkgs/development/python-modules/pyiqvia/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "bachya";
+    repo = pname;
+    rev = version;
+    sha256 = "1yjsbq4p040a35n8g2l0wgnv83wkjirv4rj73s2mjzn3cxf395bz";
+  };
+
+  nativeBuildInputs = [ poetry ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/pyiqvia/default.nix:27]: Please poke the upstream maintainers about moving to `poetry-core`. See #103325.

**Review by mweinelt — APPROVED** [review]
LGTM.

**Pull request merged**